src/main/java/com/Crowdsourcing/shootingstars/CrowdsourcingStars.java

```diff
@@ -0,0 +1,368 @@
+import net.runelite.api.NPC;
+import net.runelite.api.NullNpcID;
+import net.runelite.api.Point;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.api.events.ChatMessage;
+import net.runelite.api.events.GameObjectDespawned;
+import net.runelite.api.events.GameObjectSpawned;
+import net.runelite.api.events.GameStateChanged;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.NpcDespawned;
+import net.runelite.api.events.NpcSpawned;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.hiscore.HiscoreEndpoint;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+@Slf4j
+public class CrowdsourcingStars
+{
+	private static final String CROWDSOURCING_URL = "https://crowdsource.runescape.wiki/shooting_stars";
+	private static final MediaType JSON = MediaType.get("application/json; charset=utf-8");
+	private static final Pattern STAR_PROGRESS = Pattern.compile("This is a size-(?<tier>[0-9]+) star.* It has been mined (?<progress>[0-9]+)%.*", Pattern.CASE_INSENSITIVE);
+	private static final int CHECK_SECONDS = 60 * 1000;
+	private static final double SUBMIT_CHANCE = 0.01;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private OkHttpClient okHttpClient;
+
+	@Inject
+	private Gson gson;
+
+	private Star trackedStar = null;
+
+	private StarData lastSent = null;
+
+	private Map<Point, Long> checkedPoints = new HashMap<>();
+
+	@Subscribe
+	public void onNpcSpawned(NpcSpawned event)
+	{
+		if (event.getNpc().getId() != NullNpcID.NULL_10629)
+		{
+			return;
+		}
+
+		var npc = event.getNpc();
+		if (trackedStar == null)
+		{
+			trackedStar = new Star(client.getWorld(), npc.getWorldLocation());
+		}
+
+		trackedStar.setNpc(npc);
+	}
+
+	@Subscribe
+	public void onNpcDespawned(NpcDespawned event)
+	{
+		if (event.getNpc().getId() != NullNpcID.NULL_10629)
+		{
+			return;
+		}
+
+		if (trackedStar == null)
+		{
+			return;
+		}
+
+		trackedStar.setNpc(null);
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		checkStarProgress();
+		checkMissingStar();
+	}
+
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged event)
+	{
+		if (event.getGameState() == GameState.LOGIN_SCREEN || event.getGameState() == GameState.HOPPING)
+		{
+			reset();
+		}
+
+		if (event.getGameState() == GameState.LOADING)
+		{
+			checkedPoints.clear();
+
+			if (trackedStar != null)
+			{
+				trackedStar.setGameObject(null);
+			}
+		}
+	}
+
+	@Subscribe
+	public void onGameObjectSpawned(GameObjectSpawned event)
+	{
+		var gameObject = event.getGameObject();
+		int tier = Tiers.of(gameObject.getId());
+		if (tier < 0)
+		{
+			return;
+		}
+
+		if (trackedStar == null)
+		{
+			trackedStar = new Star(client.getWorld(), gameObject.getWorldLocation());
+		}
+
+		trackedStar.setGameObject(gameObject);
+		if (trackedStar.getTier() == tier)
+		{
+			return;
+		}
```

Comment:
Won't line 161 always eval to true since we are setting the trackedStar's game object to be the thing we got the tier from? Not sure if I'm missing something here.